### PR TITLE
Move cennznet/prml

### LINKED ISSUE #50: Move `prml/` crates back into plug-blockchain.
Currently lives here: https://github.com/cennznet/cennznet/tree/master/prml

Add this back into plug-blockchain. It does not need to be removed from CENNZnet for this task.

It was historically moved to make the development cycle simpler for CENNZnet

### PR DESCRIPTION
**Pull request by cowboy-bebug**
Discarded commit histories as they are not relevant to plug.

Edit: the following wont' be moved:
- `prml/fees`: needs to be a separate refactor task
    - relates to https://github.com/cennznet/cennznet/issues/37)
    - has dependency on https://github.com/cennznet/plug-blockchain/blob/master/srml/support/src/fee.rs
- `prml/generic-asset`: will use the upstream merged `srml/generic-asset`

Closes #50

### CODE REVIEW

**Review by jordy25519 — CHANGES_REQUESTED** [review]
Requires changes following offline discussion to fix some tech debt

File: srml/support/test/tests/reserved_keyword/on_initialize.stderr [hunk 1]
```diff
@@ -1,47 +1,47 @@
 error: Invalid call fn name: `on_finalize`, name is reserved and doesn't match expected signature, please refer to `decl_module!` documentation to see the appropriate usage, or rename it to an unreserved keyword.
   --> $DIR/on_initialize.rs:39:1
    |
-30 | reserved!(on_finalize on_initialize on_finalise on_initialise offchain_worker deposit_event);
+39 | reserved!(on_finalize on_initialize on_finalise on_initialise offchain_worker deposit_event);
```
Comment (jordy25519) [inline srml/support/test/tests/reserved_keyword/on_initialize.stderr]: Ah this is from doughnut changes, thought I had fixed this 😅

Comment (cowboy-bebug) [reply srml/support/test/tests/reserved_keyword/on_initialize.stderr]: Yep I got ur back on this one 😃

File: prml/attestation/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+//!
+//! Topic and Value are U256 integers. This means that Topic and Value can technically store any value that can be represented in 256 bits.
+//!
+//! The user of the module must convert whatever value that they would like to store into a value that can be stored as a U256.
+//!
+//! It is recommended that Topic be a string value converted to hex and stored on the blockchain as a U256.
+//! Value is typically the hash of an off-chain attestation document but can be any value that fits the application's needs.
+//!
+//!
+//! We recommend that you use the Attestation JavaScript SDK when working with this module. The SDK can be found at https://github.com/cennznet/api.js/tree/master/packages/crml-attestation
```
Comment (jordy25519) [inline prml/attestation/src/lib.rs]: I think we can remove this comment altogether

**Review by jordy25519 — CHANGES_REQUESTED** [review]

**Review by jordy25519 — APPROVED** [review]

**Comment by jordy25519** [comment]
Approval granted the CI builds check out 😄

**Pull request merged**